examples/ui/ui_texture_atlas.rs

```diff
@@ -0,0 +1,72 @@
+//! This example illustrates how to use `TextureAtlases` within ui
+
+use bevy::{prelude::*, winit::WinitSettings};
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins.set(
+            // This sets image filtering to nearest
+            // This is done to prevent textures with low resolution (e.g. pixel art) from being blurred
+            // by linear filtering.
+            ImagePlugin::default_nearest(),
+        ))
+        // Only run the app when there is user input. This will significantly reduce CPU/GPU use.
+        .insert_resource(WinitSettings::desktop_app())
+        .add_systems(Startup, setup)
+        .add_systems(Update, increment_atlas_index)
+        .run();
+}
+
+fn setup(
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+    mut texture_atlases: ResMut<Assets<TextureAtlas>>,
+) {
+    // Camera
+    commands.spawn(Camera2dBundle::default());
+
+    let texture_handle = asset_server.load("textures/rpg/chars/gabe/gabe-idle-run.png");
+    let texture_atlas =
+        TextureAtlas::from_grid(texture_handle, Vec2::new(24.0, 24.0), 7, 1, None, None);
+    let texture_atlas_handle = texture_atlases.add(texture_atlas);
+
+    // root node
+    commands
+        .spawn(NodeBundle {
+            style: Style {
+                width: Val::Percent(100.0),
+                height: Val::Percent(100.0),
+                justify_content: JustifyContent::Center,
+                align_items: AlignItems::Center,
+                ..default()
+            },
+            ..default()
+        })
+        .with_children(|parent| {
+            parent.spawn((TextureAtlasImageBundle {
+                style: Style {
+                    width: Val::Px(256.),
+                    height: Val::Px(256.),
+                    border: UiRect::all(Val::Px(2.)),
+                    ..default()
+                },
+                texture_atlas: texture_atlas_handle,
+                texture_atlas_sprite: UiTextureAtlasSprite(TextureAtlasSprite::new(0)),
+                ..default()
+            },));
```

Comment:
```suggestion
    parent.spawn(NodeBundle {
        style: Style {
            border: UiRect::all(Val::Px(2.)),
            ..default(),
        },
        ..Default::default()
    }).with_children(|parent| {
        parent.spawn((TextureAtlasImageBundle {
            style: Style {
                width: Val::Px(256.),
                height: Val::Px(256.),
                ..default()
            },
            texture_atlas: texture_atlas_handle,
            texture_atlas_sprite: UiTextureAtlasSprite(TextureAtlasSprite::new(0)),
            ..default()
        },));
    });            
```

Borders shouldn't be added to UI nodes with a `MeasureFunc` instead you should wrap them inside another node.. and I've just realised you haven't added a `MeasureFunc`. A `MeasureFunc` is needed so that the UI can automatically size the image within the space available to preserve its aspect ratio rather than users being required to specify exact dimensions.

The measure func stuff is not very well documented yet, we've only just reworked it to fix the text systems and it still has some bugs we are working out. Its used by intrinsically sized nodes containing things like images and text where the size of the node needs to take into account the size of their content and can't just be derived from the style constraints by the layout algorithm. 

You can see why borders and padding don't work if you look at the `extract_uinodes` implementation again. The image fills the entire bounds of the uinode with no space allotted for a border or padding.

Intrinsically sized nodes are also limited in that they shouldn't have any children. There is a bug at the moment in the `UI` example where the Bevy logo isn't displayed because the alt-text accessibility implementation used child nodes.

Reply:
I don't think we need borders at all for this example, I think that was left over from early tests I did and I can remove it completely